src/main/kotlin/no/nav/familie/oppdrag/simulering/SimuleringResultatTransformer.kt

```diff
@@ -0,0 +1,88 @@
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+class SimuleringResultatTransformer {
+
+    fun mapSimulering(beregning: Beregning, utbetalingsoppdrag: Utbetalingsoppdrag): DetaljertSimuleringResultat {
+        val simuleringMottakerListe: List<SimuleringMottaker> = mutableListOf()
+        for (periode in beregning.beregningsPeriode) {
+            for (stoppnivaa in periode.beregningStoppnivaa) {
+                val mottakerId = hentOrgNrEllerFnr(stoppnivaa.utbetalesTilId)
+                val requestMottakerId = hentOrgNrEllerFnr(utbetalingsoppdrag.aktoer)
+                val harSammeAktørIdSomBruker = requestMottakerId == mottakerId
+                val posteringer: List<SimulertPostering> = mutableListOf()
+                for (detaljer in stoppnivaa.beregningStoppnivaaDetaljer) {
+                    val postering: SimulertPostering = mapPostering(false, stoppnivaa, detaljer)
+                    posteringer.plus(postering)
+                }
+                val mottaker = SimuleringMottaker(simulertPostering = posteringer, mottakerNummer = beregning.gjelderId,
+                                                  mottakerType = utledMottakerType(stoppnivaa.utbetalesTilId,
+                                                                                   harSammeAktørIdSomBruker))
+                simuleringMottakerListe.plus(mottaker)
+            }
+        }
+        return DetaljertSimuleringResultat(simuleringMottakerListe)
+    }
+
+    private fun mapPostering(utenInntrekk: Boolean,
+                             stoppnivaa: BeregningStoppnivaa,
+                             detaljer: BeregningStoppnivaaDetaljer): SimulertPostering {
+        return SimulertPostering(
+                betalingType = utledBetalingType(detaljer.belop),
+                beløp = detaljer.belop,
+                fagOmrådeKode = FagOmrådeKode.fraKode(stoppnivaa.kodeFagomraade),
+                fom = parseDato(detaljer.faktiskFom),
+                tom = parseDato(detaljer.faktiskTom),
+                forfallsdato = parseDato(stoppnivaa.forfall),
+                posteringType = PosteringType.fraKode(detaljer.typeKlasse),
+                utenInntrekk = utenInntrekk)
+    }
+
+    private fun hentOrgNrEllerFnr(orgNrEllerFnr: String): String {
+        return if (erOrgNr(orgNrEllerFnr)) {
+            orgNrEllerFnr.substring(2)
+        } else {
+            orgNrEllerFnr
+        }
+    }
+
+    private fun utledMottakerType(utbetalesTilId: String, harSammeAktørIdSomBruker: Boolean): MottakerType {
+        if (harSammeAktørIdSomBruker) {
+            return MottakerType.BRUKER
+        }
+        return if (erOrgNr(utbetalesTilId)) {
+            MottakerType.ARBG_ORG
+        } else MottakerType.ARBG_PRIV
+    }
+
+    private fun erOrgNr(verdi: String): Boolean {
+        Objects.requireNonNull(verdi, "org.nr verdi er null")
```

Comment:
Veri kan ikke være null her, så denne sjekk trengs ikke. Men mulig den burde sjekkes for empty?

Reply:
fjernet requireNonNull sjekk og brukte StringUtils.nullOrEmprty sjekk i stedet